NPE on ::new method refs (Cannot invoke "org.eclipse.jdt.core.dom.IMethodBinding.isSynthetic()" because "functionalMethod" is null)

eclipse.jdt.ls version: built from master (11302ccf40b05c44fd4c8fb8ed3db913ac87272a)
java: 16

With a code snippet like this:


```java
public void foo() {
    List.of(HashMap::new);
}
```

(HashMap not being imported)

Triggering code actions if the cursor is on top of `new` triggers the following error:


```
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.jdt.core.dom.IMethodBinding.isSynthetic()" because "functionalMethod" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:645)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NullPointerException:
Cannot invoke "org.eclipse.jdt.core.dom.IMethodBinding.isSynthetic()" because "functionalMethod" is null
	at org.eclipse.jdt.ls.core.internal.text.correction.QuickAssistProcessor.getFunctionalMethodForMethodReference(QuickAssistProcessor.java:326)
	at org.eclipse.jdt.ls.core.internal.text.correction.QuickAssistProcessor.getConvertMethodReferenceToLambdaProposal(QuickAssistProcessor.java:789)
	at org.eclipse.jdt.ls.core.internal.text.correction.QuickAssistProcessor.getAssists(QuickAssistProcessor.java:175)
	at org.eclipse.jdt.ls.core.internal.handlers.CodeActionHandler.getCodeActionCommands(CodeActionHandler.java:184)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$15(JDTLanguageServer.java:649)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:75)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	... 6 more
```



The payload sent to the server:


```
{
  context = {
    diagnostics = { {
        code = "16777218",
        message = "HashMap cannot be resolved to a type",
        range = {
          end = {
            character = 21,
            line = 7
          },
          start = {
            character = 14,
            line = 7
          }
        },
        severity = 1,
        source = "Java"
      } }
  },
  range = {
    end = {
      character = 23,
      line = 7
    },
    start = <table 1>
  },
  textDocument = {
    uri = "file:///path/to/App.java"
  }
}
```

(the `<table 1>` here is a vim logging artifact, proper values got sent)

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/QuickAssistProcessor.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/QuickAssistProcessor.java
@@ -323,7 +323,7 @@ public class QuickAssistProcessor {
 		}
 
 		IMethodBinding functionalMethod = targetTypeBinding.getFunctionalInterfaceMethod();
-		if (functionalMethod.isSynthetic()) {
+		if (functionalMethod != null && functionalMethod.isSynthetic()) {
 			functionalMethod = Bindings.findOverriddenMethodInType(functionalMethod.getDeclaringClass(), functionalMethod);
 		}
 		return functionalMethod;

